draft-ietf-tls-tls13.md

```diff
@@ -2705,18 +2806,39 @@ The following rules apply to the certificates sent by the server:
                        hash algorithm that will be employed in the
                        ServerKeyShare message.  The public key
                        MUST use a curve and point format supported by
-                       the client, as described in  [RFC4492].
+                       the client, as described in [RFC4492].
 ~~~~
 
 - The "server_name" and "trusted_ca_keys" extensions {{RFC6066}} are used to
-  guide certificate selection. As servers MAY require the presence of the server_name
-  extension, clients SHOULD send this extension.
+  guide certificate selection. As servers MAY require the presence of the
+  "server_name" extension, clients MUST send this extension when applicable
+  to the current application protocol. (see {{mti-extensions}})
+
+All certificates provided by the server MUST be signed by a
+hash/signature algorithm pair that appears in the "signature_algorithms"
+extension provided by the client, where possible (see {{signature-algorithms}}).
+If the server cannot produce a certificate chain that is signed only via the
+indicated supported pairs, then it SHOULD continue the handshake by sending
+the client a certificate chain of its choice that may include algorithms
+that are not known to be supported by the client. This fallback chain MAY use
+the deprecated SHA hash algorithms, SHA-1 or SHA-224, but only if necessary.
+If the client cannot construct an acceptable chain using the provided
+certificates and decides to abort the handshake, then it MUST send an
+"unsupported_certificate" alert message and close the connection.
+
+Any endpoint receiving any certificate signed using any signature algorithm
+using an MD5 hash MUST send a "bad_certificate" alert message and close
+the connection.
```

Comment:
If you self-sign, you can upgrade to a better hash yourself. I see no reason to add any exception for MD5 now that collisions cost under a dollar, even if collisions might not be relevant in certain use-cases. Any use of MD5 comes with the risk that someone can screw up and use MD5.

Self-signed SHA1 can be used in a fallback just the same as others.